internal/image_template_test.go

```diff
@@ -188,9 +192,7 @@ var (
 
 	registryRule          = &internal.OCIImageLocation{Registry: "registry.vmware.com"}
 	repositoryPrefixRule  = &internal.OCIImageLocation{RepositoryPrefix: "my-company"}
-	digestRule            = &internal.OCIImageLocation{Digest: "sha256:eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"}
```

Comment:
> I think we're seeing the tension between a pure function versus the real use case.

I agree, although I tend to implement just enough to meet the real use-case even though It's likely as you pointed out the other day, that some of these features might come back, such as overriding an image.

> Meaning it never assumed the digest was set, or that changing a digest was a problem. Or changing a tag, or changing the name of the image itself. It never assumed that you had registry and repo-prefix set. So, it was coded to that generic specification.

I'd not call it a problem, but just unnecessary code that makes code maintenance, reviews and testing more complex than it should be imo, at least at this early stage.

> In reality, we always know the digest of the destination image. We enforce (today) that the digests will match. and we don't allow setting tags.

This diff does nothing about it, it just does a cleanup on smth that I left behind, removing the notion of overriding the digest, since we do not allow it today as you said. If, we re-enable it via force push or smth, we'll put it back.

> In that spirit, we could (should?) change the tests to match those assumptions.
I've changed to tests to remove non-supported features such as rewriting tags or digest. Am I missing any other tests?

You raised very good points and I should be more careful when I chance things, so it might be good if we could touch base so I can learn more about the initial context.